Atomic/syscontainers.py

```diff
@@ -2017,18 +2017,18 @@ def get_default_system_name(image):
 
     @staticmethod
     def _parse_imagename(imagename):
-        sep = imagename.find("/")
-        reg, image = imagename[:sep], imagename[sep + 1:]
-        if '.' not in reg:
-            # if the registry doesn't look like a domain, consider it as the
-            # image prefix
-            reg = ""
-            image = imagename
-        sep = image.find(":")
-        if sep > 0:
-            return reg, image[:sep], image[sep + 1:]
+        image, _, tag = imagename.partition(":")
```

Comment:
>  but I'm seeing a different result when port number is in the registry

ah right, did not know that will be a problem.  (thanks @ashcrow  for pointing it out and /me took down some notes).  I believe this should be the culprit here, this only takes the leftmost colon, but the tag is usually separated by the rightmost colon.

Reply:
> Good catch, will update

oops, looks like I was a bit late, you can ignore this and the previous comment ;).